js/glowingbear.js

```diff
@@ -687,7 +687,26 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         }
     };
 
+    
+    $scope.parseHash = function() {
+
+        //Fill in url parameters, they take precedence over the stored settings, but store them
+        var params = $location.$$hash.split('&').map(function(val) { return {key: val.split('=')[0], value: val.split('=')[1]}; });
+        var hostParam = params.find(function(p) { return p.key === 'host'; });
+        var portParam = params.find(function(p) { return p.key === 'port'; });
+        var pathParam = params.find(function(p) { return p.key === 'path'; });
+        var passwordParam = params.find(function(p) { return p.key === 'password'; });
+        var autoconnectParam = params.find(function(p) { return p.key === 'autoconnect'; });
+        if(hostParam) { $scope.settings.host = hostParam.value; $scope.settings.hostField = hostParam.value;}
+        if(portParam) { $scope.settings.port =  parseInt(portParam.value);}
+        if(pathParam) { $scope.settings.path = pathParam.value;}
```

Comment:
If `pathParam` is set, I think `settings.hostField` should be set to `host + ':' + port + '/' + path` so that the changed path is visible in the UI